commands/audit/scarunner.go

```diff
@@ -238,6 +247,29 @@ func GetTechDependencyTree(params xrayutils.AuditParams, tech coreutils.Technolo
 	return
 }
 
+func getCurationCacheFolderAndLogMsg(params xrayutils.AuditParams, tech coreutils.Technology) (logMessage string, curationCacheFolder string, err error) {
+	if params.IsCurationCmd() {
+		curationCacheFolder, err = getCurationCacheByTech(tech)
+		if err != nil {
+			return logMessage, curationCacheFolder, err
+		}
+
+		if isExist, err := fileutils.IsDirExists(curationCacheFolder, false); isExist {
+			if isEmpty, err := fileutils.IsDirEmpty(curationCacheFolder); !isEmpty {
+				return logMessage, curationCacheFolder, err
+			} else if err != nil {
+				return logMessage, curationCacheFolder, err
+			}
+		} else if err != nil {
+			return logMessage, curationCacheFolder, err
+		}
+
+		logMessage = ". Project's cache is currently empty, so this run may take longer to complete"
+
+	}
+	return logMessage, curationCacheFolder, err
+}
```

Comment:
```suggestion
        if !params.IsCurationCmd() {
        	return
        }
        if curationCacheFolder, err = getCurationCacheByTech(tech); err != nil {
		return
	}
	dirExist, err := fileutils.IsDirExists(curationCacheFolder, false)
	if err != nil {
		return
	}
	if dirExist {
		if dirIsEmpty, err := fileutils.IsDirEmpty(curationCacheFolder); err != nil || !dirIsEmpty {
			return
		}
	}
	logMessage = ". Project's cache is currently empty, so this run may take longer to complete"
	return
}
```
* keep func as flat as possible for readability.
* If you named to returned values, you don't need to write them on return.